### PR TITLE
Match 121

### PR DESCRIPTION
**Pull request by s3204581**
Admin and user portal implementation, various bug fixes with new listing and header

### CODE REVIEW

File: UI/zip-it/src/App.tsx [hunk 1]
```diff
@@ -302,43 +325,63 @@ function App() {
...
         console.log(result);
 
         setNewListingDisabled(false);
       });
   };
 
+  //portal type selection logic
+  function UserAdminPortalDisplay(logInDetails) {
+    if (
+      authenticated?.userEmail ===
```
Comment (mpbeach) [inline UI/zip-it/src/App.tsx:360]: I think this is an odd way to check for this, I understand it's early but the User Model has a build in Role number which you can grab and store as part of the logInDetails which can be used for this check on any account (ie. if role = 1, etc)

File: UI/zip-it/src/data/mutations.tsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ function mutation(props): DocumentNode {
     } if (props.type === "confirm") {
         result = gql`
                 mutation {
-                    confirmUser(userEmail: "${props.data.userEmail}", confirmationCode: ${props.data.confirmationCode}) {
+                    confirmUser(userEmail: "${props.data.userEmail}", confirmationCode: ${props.data.confirmationCode}, }) {
```
Comment (mpbeach) [inline UI/zip-it/src/data/mutations.tsx:21]: Can I ask why this is a mutation and not a query?

Comment (s3204581) [reply UI/zip-it/src/data/mutations.tsx]: This wasn't implemented by myself or this branch and currently resides in main.

File: UI/zip-it/src/App.tsx (hunk 1 above)
Comment (s3204581) [reply UI/zip-it/src/App.tsx:360]: Exactly, this was for testing purposes, RoleID = 1 is the proper function call for admin. I'll have to have a chat re: if there's an admin user with RoleID1 on the backend we can all log into and have suspended this PR until this is implemented.

Comment (s3204581) [reply UI/zip-it/src/App.tsx]: Testing complete, admin portal is now only accessed by RoleID via query.

**Review by mpbeach — APPROVED** [review]

**Pull request merged**